### PR TITLE
Add ChangeJavaCompatibility recipe

### PR DESCRIPTION
**Pull request by shanman190**

### CODE REVIEW

File: rewrite-gradle/src/main/java/org/openrewrite/gradle/ChangeJavaCompatibility.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+
+import java.util.List;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class ChangeJavaCompatibility extends Recipe {
+    @Option(displayName = "New version",
+            description = "The Java version to update source compatibility to",
```
Comment (sambsnyd) [inline rewrite-gradle/src/main/java/org/openrewrite/gradle/ChangeJavaCompatibility.java:24]: Consider mentioning the valid values. If someone wants Java 8 should they put in "8" or "1.8"?

**Review by sambsnyd — CHANGES_REQUESTED** [review]
Good stuff. Thorough tests. 👍

**Comment by shanman190** [comment]
Not sure about the failures. This was rebased onto the tip of `main`, but there's still some errors. 🤔

**Comment by shanman190** [comment]
Latest rebase is now working. :)

**Pull request merged**